### DISCUSSION TITLE
Support for linters and other static code analyzers

### DISCUSSION THREAD

**Discussion by gerlacdt** [helix-editor/helix#10681] [Ideas] [answered]
As far as I know helix supports LSP servers and formatters.

For an even better developer experience, an integration for linters like (golangci-lint, eslint, ruff) would be a nice addition. 
Sometimes there are wrapper LSP servers provided for such linters but they are less supported.

Currently, the only way to visualize the linter output is to switch to the terminal and run them there. It would be nice to see the linter warnings directly in helix. This would drastically shorten the feedback loop.

Are there any future plans for such an integration? For example Emacs has [flycheck](https://www.flycheck.org/en/latest/) and NeoVim has [none-ls](https://github.com/nvimtools/none-ls.nvim)

**Comment by kirawi** [comment, marked as answer]
Probably not. For language-specific tooling, we only implement established standards like LSP and DAP. Otherwise we'd have a lot of language-specific code in core. This is something that would be better as a plugin once the plugin system lands. I believe some people do use https://github.com/mattn/efm-langserver to use linters though.
